slotsmanager.c

```diff
@@ -412,8 +414,13 @@ slots_reserve(int where, int atleast[6])
     int ptype;
     int good = 1;
 
-    if(SlotsManager->Base == NULL)
+    if(SlotsManager->Base == NULL) {
         SlotsManager->Base = (char*) mymalloc("SlotsBase", 0);
+        /* This is so the ptr is never null! Avoid undefined behaviour. */
+        for(ptype = 5; ptype >= 0; ptype--) {
+            SlotsManager->info[ptype].ptr = SlotsManager->Base;
```

Comment:
Wouldn't it be better to set the ptr to NULL such that access on uninitialized .ptr  immediately lead to segfault?

Reply:
No, because of the memmove. The ptr is null at this point already, but we should not call memmove on a null pointer, even with size 0, because the compiler may optimise the pointer away.

Reply:
We are supposed to avoid memmove on a null pointer by checking `enabled` flags in `_reserve`.

Reply:
There is a hole when we reserve the first block: at that point even the enabled types are still NULL.

Reply:
I see.